packages/next/src/seo/types.ts

```diff
@@ -0,0 +1,10 @@
+import type { Metadata } from 'next';
+
+export type MetadataParser< T > = ( data: T ) => Metadata;
+
+export interface MetadataPlugin< TFrag,TData > {
+	fragmentDoc: TFrag;
+	parser: ( data: TData ) => Metadata;
+    type: "root" | "template"
```

Comment:
Assuming I understand the purpose of this field, I think it should be called `location` or something else that's more self-documenting

Reply:
Honestly I feel the others could use some work too: `fragment` instead of `fragmentDoc`, and idk if `parser` correctly defines what our metadata callback functions do.